### PR TITLE
updated TypeScript to latest 3.9.3

### PR DESCRIPTION
**Pull request by royra**
also updated jest and typescript-eslint

fixed lowerdash tests to cover index.ts

### CODE REVIEW

File: build_utils/find_conflicting_versions.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    res[name][version].push(manifest)
+  })
+  return res
+}, {})
+
+const mismatchedVersions = Object.fromEntries(
+  Object.entries(depsVersions).filter(([_dep, versions]) => Object.keys(versions).length > 1)
+)
+
+console.dir(mismatchedVersions)
```
Comment (ori-moisis) [inline build_utils/find_conflicting_versions.js:48]: should we make this part of the build and actually fail if there are conflicting versions?

File: packages/core/test/core/adapters/adapters.test.ts [hunk 2]
```diff
@@ -15,6 +15,7 @@
 */
 import { InstanceElement, ElemID, ObjectType } from '@salto-io/adapter-api'
 import * as utils from '@salto-io/adapter-utils'
+import * as utilsSource from '@salto-io/adapter-utils/dist/src/utils'
```
Comment (ori-moisis) [inline packages/core/test/core/adapters/adapters.test.ts:18]: are you sure it is OK to import something internal from a different package?
maybe the other solution of mocking all of `adapter-utils` is better in this case when we have to mock something across package boundaries?

File: build_utils/find_conflicting_versions.js (hunk 1 above)
Comment (royra) [reply build_utils/find_conflicting_versions.js]: having the same versions is a best practice, but not really required.

anyway - done.

File: packages/core/test/core/adapters/adapters.test.ts (hunk 2 above)
Comment (royra) [reply packages/core/test/core/adapters/adapters.test.ts]: replaced `jest.spyOn` with the black magic of `jest.mock`. I don't really like it but it works, and I couldn't get the other fix (`loose: true`) working.

**Pull request merged**